planning/autoware_remaining_distance_time_calculator/CMakeLists.txt

```diff
@@ -4,10 +4,22 @@ project(autoware_remaining_distance_time_calculator)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
+generate_parameter_library(remaining_distance_time_calculator_parameters
+  param/remaining_distance_time_calculator_parameters.yaml
+)
+
 ament_auto_add_library(${PROJECT_NAME} SHARED
   src/remaining_distance_time_calculator_node.cpp
 )
 
+target_link_libraries(${PROJECT_NAME}
+  remaining_distance_time_calculator_parameters
+)
+
+target_compile_options(${PROJECT_NAME} PUBLIC
```

Comment:
Actually this CMake target compile options rule caught my attention as being `PUBLIC` to suppress deprecated errors.
Using `PUBLIC` in this case is not risky but unnecessary. 
- It is not risky because the error suppression will not propagate to other targets, as other targets cannot "link" to an executable target and here we are using this command with an executable.
- It unnecessary from the sense of best practices, as `PRIVATE` is often the better choice, keeping the compile option strictly for that target. 

That is why I do really recommend to have this rule as 
```
target_compile_options(${PROJECT_NAME} PRIVATE
  -Wno-error=deprecated-declarations
)
```

Reply:
Thank you for pointing it out, I fixed it just now https://github.com/autowarefoundation/autoware.universe/pull/8826/commits/e42d4e4e4468a5bfdd3476a1468b599abc672939